### PR TITLE
Yolo optimization

### PR DESCRIPTION
**Pull request by likholat**

### CODE REVIEW

**Comment by likholat** [comment]
@dkurt

File: model-optimizer/extensions/front/YOLO.py [hunk 1]
```diff
@@ -51,12 +56,184 @@ def transform_graph(self, graph: Graph, replacement_descriptions):
...
+                result.in_port(ind).connect(node.out_port(0))
+
+            result = Transpose(graph, {}).create_node([result, tr_axes])
+
+            # if b > 1:
+
             # TODO: do we need change axis for further permutation
-            region_layer_node.dim_attrs.remove('axis')
-            Result(graph, {'name': region_layer_node.id + '/Result'}).create_node([region_layer_node])
+            result.dim_attrs.remove('axis')
```
Comment (dkurt) [inline model-optimizer/extensions/front/YOLO.py:238]: Still need to be removed from `region_layer_node`

File: model-optimizer/extensions/front/YOLO.py [hunk 2]
```diff
@@ -51,12 +59,188 @@ def transform_graph(self, graph: Graph, replacement_descriptions):
...
+
+            import numpy as np
+            from mo.ops.reshape import Reshape
+            from mo.ops.const import Const
+            from mo.ops.concat import Concat
+            from extensions.ops.transpose import Transpose
+            from extensions.ops.split import VariadicSplit
+            from extensions.ops.elementwise import Less, Mul, Greater, Sub, Add, Div
+            from extensions.ops.select import Select
+            from extensions.ops.activation_ops import Sigmoid, Exp
```
Comment (dkurt) [inline model-optimizer/extensions/front/YOLO.py:78]: Move to the top of the file

File: model-optimizer/extensions/front/YOLO.py [hunk 3]
```diff
@@ -40,7 +50,13 @@ class YoloV3RegionAddon(FrontReplacementFromConfigFileGeneral):
     """
     replacement_id = 'TFYOLOV3'
 
+
     def transform_graph(self, graph: Graph, replacement_descriptions):
+        hw = [13, 26, 52]
+        dims_reshape = [[507, 85], [2028, 85], [8112, 85]]
+        all_blobs = [10, 13, 16, 30, 33, 23, 30, 61, 62, 45, 59, 119, 116, 90, 156, 198, 373, 326]
+        all_masks = [[6,7,8], [3,4,5], [0,1,2]]
```
Comment (dkurt) [inline model-optimizer/extensions/front/YOLO.py:58]: let's replace hardcoded values to argument based

File: model-optimizer/extensions/front/YOLO.py [hunk 4]
```diff
@@ -51,12 +70,184 @@ def transform_graph(self, graph: Graph, replacement_descriptions):
...
+            thresh_node = Const(graph, {'value': thresh}).create_node()
+            mask = Greater(graph, {}).create_node([probs, thresh_node])
+
+            zero_node = Const(graph, {'value': np.zeros((int(rows - 5), cols), dtype=np.float)}).create_node()
+            probs_res = Select(graph, {}).create_node()
+            probs_res.in_port(0).get_connection().set_source(mask.out_port(0))
+            probs_res.in_port(1).get_connection().set_source(probs.out_port(0))
+            probs_res.in_port(2).get_connection().set_source(zero_node.out_port(0))
+
+            concat_shape = Const(graph, {'value': np.array([1, cols], dtype=np.float)}).create_node()
```
Comment (dkurt) [inline model-optimizer/extensions/front/YOLO.py:239]: `[1, -1]`

File: model-optimizer/extensions/front/YOLO.py [hunk 5]
```diff
@@ -40,7 +50,15 @@ class YoloV3RegionAddon(FrontReplacementFromConfigFileGeneral):
...
     replacement_id = 'TFYOLOV3'
 
+
     def transform_graph(self, graph: Graph, replacement_descriptions):
+        input_node = Node(graph, 'inputs')
+        hNorm = input_node.shape[1]
+        wNorm = input_node.shape[2]
+        hw_norm = 32
+
+        enable_postprocess = replacement_descriptions.get('postprocess', False)
```
Comment (dkurt) [inline model-optimizer/extensions/front/YOLO.py:60]: add this option to JSON with `False` value by default

**Pull request closed without merging**